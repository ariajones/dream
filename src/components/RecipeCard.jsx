import React from "react"
import { Link } from "react-router-dom"
import { deleteCardRecipe } from '../utils'

function RecipeCard ({ recipe }) {
    return (
      <>
      <div className="card">
        <Link to={{ pathname: `/recipe/${recipe.id}`, recipe: recipe}}> {/* --- DYNAMIC RECIPE ID PATH--- */}
            <div className="card-image">
              <figure className="image is-5by1">
                <img src={recipe.imagePath} alt={recipe.name}/> {/* --- IMAGE PATH --- */}
              </figure>
            </div>
              <div className="card-content">
                <div className="media mb-0">
                  <div className="media-content">
                    <p className="title is-5 mb-0">{recipe.name}</p> {/* --- NAME OF RECIPE --- */}
                    <div className="field is-grouped">
                      <p className="control">
                        <Link to={`/recipe/${recipe.id}`}><button className="button is-link is-small is-rounded">
                          <i className="far fa-calendar-plus"></i>
                        </button>
                        </Link>
                      </p>
                      <p className="control">
                        <Link to={`/recipe/edit/${recipe.id}`}><button className="button is-small is-rounded is-primary">
                        <i class="far fa-edit"></i>
                        </button>
                        </Link>
                      </p>
                      <p className="control">
                      <Link to='/recipes'>
                      <button onClick={() => deleteCardRecipe(recipe)}className="button is-danger is-small is-rounded">
                        <i className="fas fa-trash-alt"></i>
                        </button>
                      </Link>
                      </p>
                    </div>
                  </div>
                </div>

                <div className="content">
                  <p className="mb-0">Serves: {recipe.serves}</p> {/* --- SERVES --- */}
                  <p>Prep time: {recipe.prepTime}</p> {/* --- PREP TIME --- */}
                </div>
            </div>
        </Link>
      </div>
            
      <div className="seperation"></div>
    </>
  )
}

export default RecipeCard