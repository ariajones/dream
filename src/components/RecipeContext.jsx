import React, { useState, createContext, useEffect } from 'react'
import firebase from 'firebase/app'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase
      .firestore()
      .collection('recipes')
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}
