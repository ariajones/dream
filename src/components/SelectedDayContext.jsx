import React, { useState, createContext } from 'react'

export const SelectedDayContext = createContext()

export const SelectedDayProvider = ({ children }) => {
  const [selectedDay, setSelectedDay] = useState('monday')

  return (
    <SelectedDayContext.Provider value={[selectedDay, setSelectedDay]}>
      {children}
    </SelectedDayContext.Provider>
  )
}