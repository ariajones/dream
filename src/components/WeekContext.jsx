import React, { useState, createContext, useEffect } from 'react'

import firebase from 'firebase/app'

export const WeekContext = createContext()

export const WeekProvider = ({ children }) => {
  const [week, setWeek] = useState({
    monday: '',
    tuesday: '',
    wednesday: '',
    thursday: '',
    friday: '',
    saturday: '',
    sunday: ''
  })
 

  useEffect(() => {
    firebase
      .firestore()
      .collection('week')
      .onSnapshot(snapshot => {
        const newWeek = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setWeek(newWeek[0])
      })
  },[])

  return (
    <WeekContext.Provider value={[week, setWeek]}>
      {children}
    </WeekContext.Provider>
  )
}