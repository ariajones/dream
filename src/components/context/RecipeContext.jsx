import React, { useState, createContext, useEffect, useContext } from 'react'
import firebase from 'firebase/app'
import 'firebase/auth'

import { UserContext } from './UserContext'

export const RecipeContext = createContext()

export const RecipeProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [recipes, setRecipes] = useState([])

  useEffect(() => {
    const unsubscribe = firebase 
      .firestore()
      .collection('recipes')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const newRecipes = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setRecipes(newRecipes)
      })

    return () => unsubscribe()
  }, [user])

  return (
    <RecipeContext.Provider value={[recipes, setRecipes]}>
      {children}
    </RecipeContext.Provider>
  )
}