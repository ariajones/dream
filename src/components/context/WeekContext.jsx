import React, { useState, createContext, useEffect, useContext } from 'react'
import firebase from 'firebase/app'

import { UserContext } from './UserContext'

export const WeekContext = createContext()

export const WeekProvider = ({ children }) => {
  const {user} = useContext(UserContext)
  const [week, setWeek] = useState({
    monday: '',
    tuesday: '',
    wednesday: '',
    thursday: '',
    friday: '',
    saturday: '',
    sunday: '',
    userId: ''
  })

  useEffect(() => {
    const unsubscribe = firebase 
      .firestore()
      .collection('week')
      .where('userId', '==', user ? user.uid : null)
      .onSnapshot(snapshot => {
        const newWeek = snapshot.docs.map(doc => ({
          id: doc.id,
          ...doc.data()
        }))
        setWeek(newWeek[0])
      })

    return () => unsubscribe()
  }, [user])

  return (
    <WeekContext.Provider value={[week, setWeek]}>
      {children}
    </WeekContext.Provider>
  )
}
