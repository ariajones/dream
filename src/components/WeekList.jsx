import React, { useContext } from 'react'
import { Link } from "react-router-dom"
import { WeekContext } from './WeekContext'
import { RecipeContext } from './RecipeContext'
import RecipeCard from './RecipeCard'
import { clearWeekDayAssignments } from '../utils'

function WeekList() {
  const [week] = useContext(WeekContext)
  const [recipes] = useContext(RecipeContext)
  const mondayRecipe = recipes.find(x => x.id === week.monday)
  const tuesdayRecipe = recipes.find(x => x.id === week.tuesday)
  const wednesdayRecipe = recipes.find(x => x.id === week.wednesday)
  const thursdayRecipe = recipes.find(x => x.id === week.thursday)
  const fridayRecipe = recipes.find(x => x.id === week.friday)
  const saturdayRecipe = recipes.find(x => x.id === week.saturday)
  const sundayRecipe = recipes.find(x => x.id === week.sunday)

  function clickHandler () {
    clearWeekDayAssignments()
  }

  return (
    <div className="WeekList">
      <div>
        <h2 className="subtitle is-5 has-text-centered">Your list of days and meals below:</h2><button onClick={clickHandler}>Clear Week Assignments</button>
        <br></br>
        <h3 className="title is-3 has-text-centered">Monday</h3>
        {mondayRecipe ? <RecipeCard recipe={mondayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Monday</button></Link>}
        <h3 className="title is-3 has-text-centered">Tuesday</h3>
        {tuesdayRecipe ? <RecipeCard recipe={tuesdayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Tuesday</button></Link>}
        <h3 className="title is-3 has-text-centered">Wednesday</h3>
        {wednesdayRecipe ? <RecipeCard recipe={wednesdayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Wednesday</button></Link>}
        <h3 className="title is-3 has-text-centered">Thursday</h3>
        {thursdayRecipe ? <RecipeCard recipe={thursdayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Thursday</button></Link>}
        <h3 className="title is-3 has-text-centered">Friday</h3>
        {fridayRecipe ? <RecipeCard recipe={fridayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Friday</button></Link>}
        <h3 className="title is-3 has-text-centered">Saturday</h3>
        {saturdayRecipe ? <RecipeCard recipe={saturdayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Saturday</button></Link>}
        <h3 className="title is-3 has-text-centered">Sunday</h3>
        {sundayRecipe ? <RecipeCard recipe={sundayRecipe} /> : <Link to='/recipes'><button>Add Recipe For Sunday</button></Link>}
      </div>
    </div>
  )
}

export default WeekList
