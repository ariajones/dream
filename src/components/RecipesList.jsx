import React, { useContext, useState } from 'react'

import { RecipeContext } from './context/RecipeContext'
import RecipeCard from './RecipeCard'
import Seed from './Seed'

import { sortRecipes } from '../utils.js'

function RecipesList () {
  const [recipes] = useContext(RecipeContext)
  const [sortBy, setSortBy] = useState('NAME_ASC')
  const sortedRecipes = sortRecipes([...recipes], sortBy)

  return (
    <>
      <div className="RecipesList">
        <Seed /> {/* --- TO BE DELETED FROM PRODUCTION --- */}
        <div className='recipeSort'>
          <label>Sort By:</label>{' '}
          <select value={sortBy} onChange={evt => setSortBy(evt.target.value)}>
            <option value='NAME_ASC'>Name (A-Z)</option>
            <option value='NAME_DESC'>Name (Z-A)</option>
          </select>
        </div>
        <ul>
          {sortedRecipes.map(recipe =>
            <li key={recipe.id}>
              <RecipeCard recipe={recipe} />
            </li>
          )}
        </ul>
      </div>
    </>
  )
}

export default RecipesList
