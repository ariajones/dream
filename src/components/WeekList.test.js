import React, { useState, useContext } from 'react'
import renderer from 'react-test-renderer'
import { HashRouter } from 'react-router-dom'

import WeekList from './WeekList'

import { WeekContext } from './context/WeekContext'
import { RecipeContext } from './context/RecipeContext'
import { SelectedDayContext } from './context/SelectedDayContext'
import RecipeCard from './RecipeCard'
import { clearShoppingList, clearWeekDayAssignments } from '../utils'
import RecipesList from './RecipesList'

export const FakeRecipeProvider = ({ children }) => {
    const [recipes, setRecipes] = useState([{
    id: "MOqgTKsRp9OwepdSupbr",
    method: ["Preheat the oven to 200ºC/400ºF/gas 6. Peel 1 oni…"],
    serves: "4",
    prepTime: "40 Minutes - Not too tricky",
    name: "Roasted black bean burgers",
    ingredients: ["1½ red onions", "100 g rye bread", "ground corian…"],
    imagePath: "https://imagesvc.meredithcorp.io/v3/mm/image?url=https%3A%2F%2Fimages.media-allrecipes.com%2Fuserphotos%2F5858755.jpg"
    }])
  
    return (
        <RecipeContext.Provider value={[recipes, setRecipes]}>
          {children}
        </RecipeContext.Provider>
      )
}

export const FakeWeekProvider = ({ children }) => {  
    const [week, setWeek] = useState([{
        monday: '',
        tuesday: '',
        wednesday: 'MOqgTKsRp9OwepdSupbr',
        thursday: '',
        friday: '',
        saturday: '',
        sunday: ''
    }])

    return (
        <WeekContext.Provider value={[week, setWeek]}>
            {children}
        </WeekContext.Provider>
    )
}

export const FakeSelectedDayProvider = ({ children }) => {
    const [selectedDay, setSelectedDay] = useState('monday')

    return (
        <SelectedDayContext.Provider value={[selectedDay, setSelectedDay]}>
            {children}
        </SelectedDayContext.Provider>
    )

}

it ('renders component correctly', () => {
    const dream = renderer
    .create(
        <HashRouter>
            <FakeRecipeProvider>
                <FakeWeekProvider>
                    <FakeSelectedDayProvider>
                        <WeekList>
                            <RecipeCard/>
                        </WeekList>
                    </FakeSelectedDayProvider>
                </FakeWeekProvider>
            </FakeRecipeProvider>
        </HashRouter>
    )
    .toJSON()
expect(dream).toMatchSnapshot()
})