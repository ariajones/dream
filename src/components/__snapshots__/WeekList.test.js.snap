// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders component correctly 1`] = `
<div>
  <div>
    <hr />
    <div
      className="weeklist"
    >
      <div
        className="clear-week"
      >
        <button
          className="button is-small is-rounded is-outline is-centered"
          onClick={[Function]}
        >
          Clear Week
        </button>
         
      </div>
      <div
        className="monday-clear"
      >
        <a
          href="#/recipes"
          onClick={[Function]}
        >
          <div
            className="list"
          >
            <h3
              className="subtitle is-5"
              onClick={[Function]}
            >
              Monday+ 
              <hr />
            </h3>
          </div>
        </a>
      </div>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Tuesday+
            <hr />
          </h3>
        </div>
      </a>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Wednesday+
            <hr />
          </h3>
        </div>
      </a>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Thursday+
            <hr />
          </h3>
        </div>
      </a>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Friday+
            <hr />
          </h3>
        </div>
      </a>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Saturday+
            <hr />
          </h3>
        </div>
      </a>
      <a
        href="#/recipes"
        onClick={[Function]}
      >
        <div
          className="list"
        >
          <h3
            className="subtitle is-5"
            onClick={[Function]}
          >
            Sunday+
            <hr />
          </h3>
        </div>
      </a>
    </div>
  </div>
</div>
`;
